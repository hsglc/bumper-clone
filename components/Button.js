/** @format */

import { useRouter } from "next/router";

import { RightArrowIcon } from "../icons";

const Button = ({
  children,
  icon = <RightArrowIcon />,
  route = "",
  type = "",
  id = "",
}) => {
  const router = useRouter();

  return (
    <button
      data-testid={id}
      type={type}
      onClick={() => route && router.replace(route)}
      className="group w-full flex items-center justify-center gap-2.5 h-[45px] py-2.5 border border-[#1B1B1B] rounded-[27px] bg-[#32BE50] active:bg-[#1B1B1B]">
      <span className="text-black transition-all group-hover:text-white group-active:text-white">
        {children}
      </span>
      <i> {icon}</i>
    </button>
  );
};

export default Button;
