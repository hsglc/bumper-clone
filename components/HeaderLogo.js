/** @format */

const HeaderLogo = () => {
  return (
    <div className="z-50 font-black w-min text-xl py-[8px] px-[14px] bg-black text-white italic rounded-[20px]">
      BUMPER
    </div>
  );
};

export default HeaderLogo;
