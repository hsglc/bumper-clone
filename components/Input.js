/** @format */

import TextWithIcon from "./Text";
import { QuotationMarkIcon, SuccessCheckIcon } from "../icons";

const Input = ({
  id,
  type = "text",
  labelText = "",
  value = "",
  onChange = () => {},
  placeholder = "",
  icon = null,
  isSuccessfullyValidated = null
}) => {
  const iconType =
    isSuccessfullyValidated === true ? (
      <SuccessCheckIcon />
    ) : isSuccessfullyValidated === false ? (
      <QuotationMarkIcon />
    ) : null;
  const borderStyle =
    isSuccessfullyValidated === true
      ? "border-green-500"
      : isSuccessfullyValidated === false
      ? "border-red-500"
      : "border-[#545454]/50";
  const classesOfInputTypes = {
    text: `h-12 rounded-[27px] border  ${borderStyle}`,
  };
  return (
    <label
      htmlFor={id}
      className="w-full relative flex flex-col gap-2 text-[#1B1B1B]">
      {<TextWithIcon text={labelText} icon={icon} />}
      <input
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        className={`w-full px-4 py-3 outline-none text-[#1B1B1B] placeholder:font-medium placeholder:text-[#737373]/30 ${classesOfInputTypes[type]}`}
        id={id}
        type={type}
      />
      <div className="absolute top-[60px] right-6">{iconType}</div>
    </label>
  );
};

export default Input;
