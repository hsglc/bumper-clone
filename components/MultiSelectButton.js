/** @format */

import { PlusIcon, SuccessCheckIcon } from "../icons";

const MultiSelectButton = ({
  children,
  isSelected = false,
  onToggle = () => {},
}) => {
  return (
    <div
      onClick={onToggle}
      className={`bg-white  border cursor-pointer border-black rounded-[27px] transition-all duration-300 ${
        isSelected ? "bg-black" : "hover:bg-[#DCE6E6]"
      } px-4 py-3 text-black flex items-center gap-1`}>
      <span className={isSelected && "text-white"}>{children}</span>
      {isSelected ? <SuccessCheckIcon /> : <PlusIcon />}
    </div>
  );
};

export default MultiSelectButton;
