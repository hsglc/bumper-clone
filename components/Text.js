/** @format */

const TextWithIcon = ({ text = "", icon = null, order = "fromLeft" }) => {
  return (
    <div className={`flex-center gap-1.5`}>
      {order === "fromLeft" && (
        <i className="w-5 flex justify-center">{icon}</i>
      )}
      <span className="font-black text-[#1B1B1B]">{text}</span>
      {order === "fromRight" && (
        <i className="w-5 flex justify-center">{icon}</i>
      )}
    </div>
  );
};

export default TextWithIcon;
