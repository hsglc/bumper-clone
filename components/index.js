/** @format */

export { default as TextWithIcon } from "./Text.js";
export { default as Input } from "./Input.js";
export { default as Button } from "./Button.js";
export { default as HeaderLogo } from "./HeaderLogo.js";
export { default as Header } from "./Header.js";
export { default as MultiSelectButton } from "./MultiSelectButton";
export { default as RegisterForm } from "./RegisterForm";
