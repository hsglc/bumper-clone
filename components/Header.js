/** @format */

import { useState } from "react";
import Link from "next/link";
import { useMediaQuery } from "react-responsive";

import { HeaderLogo, TextWithIcon } from "../components";
import { LoginIcon, ServiceIcon } from "../icons";

const Header = () => {
  const [isOpen, setIsOpen] = useState(false);
  const isLgScreen = useMediaQuery({
    query: "(min-width: 1024px)",
  });

  const leftSideLinks = (
    <div className={` bg-[#FF733C] `}>
      <ul
        className={`text-gray-600 text-lg tracking-wide flex flex-col lg:flex-row`}>
        <li className="navbar-leftside-elements">
          <Link href="/">
            <a className="pl-8 lg:pl-0">Apply</a>
          </Link>
        </li>
        <li className="navbar-leftside-elements">
          <Link href="/works">
            <a className="pl-8">How it works</a>
          </Link>
        </li>
        <li className="navbar-leftside-elements">
          <Link href="/about">
            <a className="pl-8">About Us</a>
          </Link>
        </li>
        <li className="navbar-leftside-elements">
          <Link href="/blog">
            <a className="pl-8">Blog</a>
          </Link>
        </li>
      </ul>
    </div>
  );

  return (
    <div className="lg:bg-[#5A698C] lg:p-6">
      <header className="layout-container py-5 lg:py-1.5 bg-[#FF733C] flex justify-between items-center lg:rounded-r-full lg:rounded-l-full">
        {isLgScreen ? leftSideLinks : <HeaderLogo />}

        {!isLgScreen && (
          <nav>
            <div
              onClick={() => setIsOpen((prevVal) => !prevVal)}
              className="navbar-icon">
              <div
                className={`navbar-icon-slices  ${
                  isOpen ? "-rotate-45  absolute" : "rotate-0"
                }`}
              />
              <div
                className={`navbar-icon-slices ${
                  isOpen ? "opacity-0" : "opacity-100"
                } `}
              />

              <div
                className={`navbar-icon-slices ${
                  isOpen ? "rotate-45  absolute" : "rotate-0"
                }`}
              />
            </div>
          </nav>
        )}
        {isLgScreen && <HeaderLogo />}
        {isLgScreen && (
          <div className="flex items-center gap-3">
            <TextWithIcon
              text="Partner sign up"
              order="fromRight"
              icon={<ServiceIcon fill="black" />}
            />
            <div className="py-1.5 px-2.5 border border-black rounded-full cursor-pointer transition-all hover:bg-white">
              <TextWithIcon
                text="Partner Login"
                order="fromRight"
                icon={<LoginIcon />}
              />
            </div>
          </div>
        )}
      </header>
      {isOpen && leftSideLinks}
    </div>
  );
};

export default Header;
