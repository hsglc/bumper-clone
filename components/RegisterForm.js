/** @format */

import Link from "next/link";

import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";

import useRegister from "../hooks/useRegister";
import { Input, MultiSelectButton, TextWithIcon, Button } from "../components";
import {
  ServiceIcon,
  EmailIcon,
  CompanyIcon,
  PhoneNumberIcon,
  HomeIcon,
  NameIcon,
} from "../icons";

const RegisterForm = () => {
  const {
    name,
    setName,
    company,
    setCompany,
    phone,
    setPhone,
    email,
    setEmail,
    postCode,
    setPostCode,
    payLater,
    setPayLater,
    payNow,
    setPayNow,
    onSubmit,
  } = useRegister();

  const schema = yup
    .object({
      name: yup.string().max(255).required(),
      company: yup.string().max(255).required(),
      phone: yup.string().required(),
      email: yup.string().min(5).max(255).email().required(),
      postCode: yup.string().max(30).required(),
    })
    .required();

  const {
    register,
    formState: { errors },

    handleSubmit,
  } = useForm({
    mode: "onBlur",
    reValidateMode: "onChange",
    resolver: yupResolver(schema),
  });

  const inputs = [
    {
      id: "name",
      icon: <NameIcon />,
      labelText: "Name",
      placeholder: "Enter your name",
      value: name,
      onChange: (e) => setName(e.target.value),
    },
    {
      id: "company",
      icon: <CompanyIcon />,
      labelText: "Company",
      placeholder: "Enter your company",
      value: company,
      onChange: (e) => setCompany(e.target.value),
    },
    {
      id: "phone",
      icon: <PhoneNumberIcon />,
      labelText: "Mobile phone number",
      placeholder: "Enter your mobile phone number",
      value: phone,
      onChange: (e) => setPhone(e.target.value),
    },
    {
      id: "email",
      icon: <EmailIcon />,
      labelText: "Email address",
      placeholder: "Enter your email address",
      value: email,
      onChange: (e) => setEmail(e.target.value),
    },
    {
      id: "postCode",
      icon: <HomeIcon />,
      labelText: "Post code",
      placeholder: "Enter your post code",
      value: postCode,
      onChange: (e) => setPostCode(e.target.value),
    },
  ];

  return (
    <form
      className="rounded-[30px] py-6 px-4 bg-white mt-4"
      onSubmit={handleSubmit(onSubmit)}>
      <h2 className="font-black">Join our network</h2>
      <p>Free to join, no monthly fees</p>

      <div className="flex flex-col gap-7 mt-4">
        {inputs.map(({ id, icon, labelText, placeholder, value, onChange }) => {
          return (
            <div key={id}>
              <Input
                register={register}
                required={true}
                key={id}
                id={id}
                icon={icon}
                labelText={labelText}
                placeholder={placeholder}
                value={value}
                onChange={onChange}
              />
              {errors[id] && (
                <p data-testid={id} className="font-bold text-red-500 mt-1">
                  {errors[id].message}
                </p>
              )}
            </div>
          );
        })}
        <div>
          <TextWithIcon
            text="What services are you interested in?"
            icon={<ServiceIcon />}
          />
          <p className="text-[#737373]">
            Please select the services youâ€™re interested in offering your
            customers
          </p>
        </div>
      </div>
      <div className="flex items-center gap-3 mt-2 mb-3">
        <MultiSelectButton
          id="payLater"
          isSelected={payLater}
          onToggle={() => setPayLater((prevState) => !prevState)}>
          PayLater
        </MultiSelectButton>

        <MultiSelectButton
          id="payNow"
          isSelected={payNow}
          onToggle={() => setPayNow((prevState) => !prevState)}>
          PayNow
        </MultiSelectButton>
      </div>

      <Button id="submitBtn">Register</Button>
      <div className="text-center mt-4">
        <Link href="/">
          <a>
            Already registered?
            <span className="login-link"> Login</span>
          </a>
        </Link>
      </div>
    </form>
  );
};

export default RegisterForm;
