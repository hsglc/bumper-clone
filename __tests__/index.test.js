/** @format */

import { render, screen, fireEvent } from "@testing-library/react";
import { HeaderLogo, RegisterForm } from "../components";
import { act } from "react-dom/test-utils";

test("whether HeaderLogo component is present", () => {
  render(<HeaderLogo />);
  const headerLogoEl = screen.getByText(/BUMPER/i);
  expect(headerLogoEl).toBeInTheDocument();
});

describe("SignIn", () => {
  describe("with valid inputs", () => {
    it("calls the onSubmit function", async () => {
      const { getByLabelText, getByRole } = render(<RegisterForm />);

      await act(async () => {
        fireEvent.change(getByLabelText("Email Address *"), {
          target: { value: "email@test.com" },
        });
        fireEvent.change(getByLabelText("Password *"), {
          target: { value: "1234567" },
        });
      });

      await act(async () => {
        fireEvent.click(getByRole("button"));
      });

      expect(mockOnSubmit).toHaveBeenCalled();
    });
  });
});
