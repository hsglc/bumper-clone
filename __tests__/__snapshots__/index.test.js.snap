// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header 1`] = `
<div>
  <div
    class="z-50 font-black w-min text-xl py-[8px] px-[14px] bg-black text-white italic rounded-[20px]"
  >
    BUMPER
  </div>
</div>
`;
